pkg/sql/pgwire/conn.go

```diff
@@ -311,11 +312,25 @@ func (c *conn) serveImpl(
 	if sqlServer != nil {
 		wg.Add(1)
 		go func() {
+			defer func() {
+				if sqlServer.GetExecutorConfig().TestingKnobs.CatchPanics {
+					if r := recover(); r != nil {
+						// Catch the panic and return it to the client as an error.
+						err := pgerror.NewErrorf(pgerror.CodeCrashShutdownError, "caught fatal error: %v", r)
+						err.Detail = string(debug.Stack())
+						writeErr(err, &c.msgBuilder, &c.writerState.buf)
+						_ /* n */, _ /* err */ = c.writerState.buf.WriteTo(c.conn)
+						c.stmtBuf.Close()
+						// Send a ready for query to make sure the client can react.
+						c.bufferReadyForQuery('I')
+						cancelConn()
```

Comment:
It looks like cancelConn won't be called if CatchPanics is false. Since it was always called before, I think it should be called if it is false.

Reply:
Good catch, done.